fairscale/experimental/wgit/pygit.py

```diff
@@ -129,3 +125,19 @@ def _path(self) -> str:
     def status(self) -> None:
         """Show the status of the git repo"""
         print(self.repo.status())
+
+    def _set_author_config(self, name: str, email: str) -> Tuple[str, str]:
+        """Set the name and email for the pygit repo collecting from the gitconfig.
+        If not available in gitconfig, set the values from the passed arguments."""
+        gitconfig = Path("~/.gitconfig").expanduser()
+        # parse the .gitconfig file for name and email
+        try:
+            data = gitconfig.read_text().split()
```

Comment:
does pygit provide a way to read it? Reading and parsing the config file directly may not be reliable.

Reply:
Yeah, I was thinking the same thing as well, this is quite hacky. However, I could not find a way from PyGit's documentation after spending some time/searching. I will take a closer look.